packages/typespec-client-generator-core/src/package.ts

```diff
@@ -524,45 +523,63 @@ function getClientDefaultApiVersion(
   return defaultVersion;
 }
 
-function getSdkInitializationType(
+function createSdkInitializationType(
   context: TCGCContext,
   client: SdkClient | SdkOperationGroup,
 ): [SdkInitializationType, readonly Diagnostic[]] {
   const diagnostics = createDiagnosticCollector();
-  let initializationModel = getClientInitialization(context, client.type);
-  let clientParams = context.__clientToParameters.get(client.type);
-  if (!clientParams) {
-    clientParams = [];
-    context.__clientToParameters.set(client.type, clientParams);
-  }
-  const access = client.kind === "SdkClient" ? "public" : "internal";
-  if (initializationModel) {
-    for (const prop of initializationModel.properties) {
-      clientParams.push(prop);
-    }
-    initializationModel.access = access;
-  } else {
-    const namePrefix = client.kind === "SdkClient" ? client.name : client.groupPath;
-    const name = `${namePrefix.split(".").at(-1)}Options`;
-    initializationModel = {
-      __raw: client.service,
-      doc: "Initialization class for the client",
-      kind: "model",
-      properties: [],
-      name,
-      isGeneratedName: true,
-      access,
-      usage: UsageFlags.Input,
-      crossLanguageDefinitionId: `${getNamespaceFullName(client.service.namespace!)}.${name}`,
-      clientNamespace: getClientNamespace(context, client.type),
-      apiVersions: context.__tspTypeToApiVersions.get(client.type)!,
-      decorators: [],
-    };
-  }
+  const initializationOptions = getClientInitialization(context, client.type);
+
+  // Create an empty client initialization model.
+  const name = `${client.kind === "SdkClient" ? client.name : getLibraryName(context, client.type)}Options`;
+  const initializationModel: SdkInitializationType = {
+    __raw: client.service,
+    doc: "Initialization class for the client",
+    kind: "model",
+    properties: [],
+    name,
+    isGeneratedName: true,
+    access: initializationOptions?.access ?? (client.kind === "SdkClient" ? "public" : "internal"),
+    usage: UsageFlags.Input,
```

Comment:
How can emitter filter out this model?

Reply:
it is not included in the `sdkPackage.models` and is deprecated now.